### PR TITLE
Implement APIs missing on Windows.

### PR DESCRIPTION
**Pull request by KubaO**
Finish implementations of missing V3Os APIs on Windows. Add V3Os::usleep. Clean up a bit.

### CODE REVIEW

File: src/V3Os.cpp [hunk 1]
```diff
@@ -220,23 +252,36 @@ vluint64_t V3Os::rand64(vluint64_t* statep) {
...
+#if defined(_WIN32) || defined(__MINGW32__)
+    NTSTATUS hr = BCryptGenRandom(NULL, reinterpret_cast<BYTE*>(data), size, BCRYPT_USE_SYSTEM_PREFERRED_RNG);
+    if (!BCRYPT_SUCCESS(hr)) {
+        v3fatal("Could not acquire random data.");
+    }
+#else
     std::ifstream is ("/dev/urandom", std::ios::in | std::ios::binary);
-    char bytes[size];
-    if (!is.read(bytes, size)) {
+    if (!is.read(data, size)) {
```
Comment (wsnyder) [inline src/V3Os.cpp]: Codacy Issue found: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042)

**Comment by wsnyder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042&bid=15757479)

File: src/V3Os.cpp [hunk 2]
```diff
@@ -266,3 +317,12 @@ uint64_t V3Os::memUsageBytes() {
     return (text + data) * getpagesize();
 #endif
 }
+
+void V3Os::usleep(int64_t usec) {
+#if defined(_WIN32) || defined(__MINGW32__)
+    std::this_thread::sleep_for(std::chrono::microseconds(usec));
+#else
+    // cppcheck-suppress obsoleteFunctionsusleep
+    ::usleep(usec);
```
Comment (wsnyder) [inline src/V3Os.cpp]: Codacy Issue found: [This C routine is considered obsolete (as opposed to the shell command by the same name).   The interaction of this function with SIGALRM and other timer functions such as sleep(), alarm(), setitimer(), and nanosleep() is unspecified (CWE-676).  Use nanosleep(2) or setitimer(2) instead. ](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042)

File: src/V3Os.cpp [hunk 3]
```diff
@@ -266,3 +317,12 @@ uint64_t V3Os::memUsageBytes() {
     return (text + data) * getpagesize();
 #endif
 }
+
+void V3Os::usleep(int64_t usec) {
```
Comment (wsnyder) [inline src/V3Os.cpp]: Codacy Issue found: [This C routine is considered obsolete (as opposed to the shell command by the same name).   The interaction of this function with SIGALRM and other timer functions such as sleep(), alarm(), setitimer(), and nanosleep() is unspecified (CWE-676).  Use nanosleep(2) or setitimer(2) instead. ](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042)

File: src/V3File.cpp [hunk 4]
```diff
@@ -447,8 +446,7 @@ class VInFilterImp {
                      || errno == EWOULDBLOCK
 #endif
                 ) {
-                // cppcheck-suppress obsoleteFunctionsusleep
-                checkFilter(false); usleep(1000); continue;
+                checkFilter(false); V3Os::usleep(1000); continue;
```
Comment (wsnyder) [inline src/V3File.cpp]: Codacy Issue found: [This C routine is considered obsolete (as opposed to the shell command by the same name).   The interaction of this function with SIGALRM and other timer functions such as sleep(), alarm(), setitimer(), and nanosleep() is unspecified (CWE-676).  Use nanosleep(2) or setitimer(2) instead. ](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042)

File: src/V3Os.h [hunk 5]
```diff
@@ -58,7 +58,8 @@ class V3Os {
     static vluint64_t rand64(vluint64_t* statep);
     static string trueRandom(size_t size);
 
-    // METHODS (performance)
+    // METHODS (time & performance)
+    static void usleep(int64_t usec);  ///< Sleep for a given number of microseconds.
```
Comment (wsnyder) [inline src/V3Os.h]: Codacy Issue found: [This C routine is considered obsolete (as opposed to the shell command by the same name).   The interaction of this function with SIGALRM and other timer functions such as sleep(), alarm(), setitimer(), and nanosleep() is unspecified (CWE-676).  Use nanosleep(2) or setitimer(2) instead. ](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042)

**Comment by wsnyder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042&bid=15757479)

File: src/V3File.cpp (hunk 4 above)
Comment (wsnyder) [inline src/V3File.cpp]: Codacy Issue found: [This C routine is considered obsolete (as opposed to the shell command by the same name).   The interaction of this function with SIGALRM and other timer functions such as sleep(), alarm(), setitimer(), and nanosleep() is unspecified (CWE-676).  Use nanosleep(2) or setitimer(2) instead. ](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042)

File: src/V3Os.cpp [hunk 6]
```diff
@@ -266,3 +319,12 @@ uint64_t V3Os::memUsageBytes() {
     return (text + data) * getpagesize();
 #endif
 }
+
+void V3Os::u_sleep(int64_t usec) {
+#if defined(_WIN32) || defined(__MINGW32__)
+    std::this_thread::sleep_for(std::chrono::microseconds(usec));
+#else
+    // cppcheck-suppress obsoleteFunctionsusleep
+    ::usleep(usec);
```
Comment (wsnyder) [inline src/V3Os.cpp]: Codacy Issue found: [This C routine is considered obsolete (as opposed to the shell command by the same name).   The interaction of this function with SIGALRM and other timer functions such as sleep(), alarm(), setitimer(), and nanosleep() is unspecified (CWE-676).  Use nanosleep(2) or setitimer(2) instead. ](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042)

**Comment by wsnyder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042&bid=15757479)

File: src/V3File.cpp (hunk 4 above)
Comment (wsnyder) [inline src/V3File.cpp]: Codacy Issue found: [This C routine is considered obsolete (as opposed to the shell command by the same name).   The interaction of this function with SIGALRM and other timer functions such as sleep(), alarm(), setitimer(), and nanosleep() is unspecified (CWE-676).  Use nanosleep(2) or setitimer(2) instead. ](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042)

**Comment by wsnyder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042&bid=15757479)

**Comment by wsnyder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042&bid=15757479)

**Comment by wsnyder** [comment]
Seems reasonable once the regressions pass.

**Comment by wsnyder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042&bid=15757479)

**Comment by KubaO** [comment]
It was a genuine bug. Yay for tests. I need to start running them myself :)

**Comment by wsnyder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042&bid=15757479)

**Comment by KubaO** [comment]
BTW the last push was a rebase, even though it applied cleanly either way.

**Comment by wsnyder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/wsnyder/verilator/pullRequest?prid=4719042&bid=15757479)

**Comment by KubaO** [comment]
Rebased. I have nothing else to add to this; feel free to merge if it's good enough.

**Comment by wsnyder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/wsnyder/verilator/pullRequest?prid=4719042&bid=15757479)

**Comment by wsnyder** [comment]
Looks good, thanks for your improvements.

**Pull request merged**